### ISSUE TITLE
[BUG] Scanning UWP apps crashes since UIAutomation does not find any child elements

### ISSUE THREAD

**Issue by chingucoding** [microsoft/axe-windows#730]
## Describe the bug

When scanning a UWP app, calling `Scan(all)` crashes with an exception originating from TargetElementLocator.cs:

```
Axe.Windows.Automation.AxeWindowsAutomationException: ERROR Automation003: Unhandled Exception System.ArgumentException: The argument must be a non-empty IEnumerable (Parameter 'desktopElements')
   at Axe.Windows.Automation.TargetElementLocator.GetA11YElementsFromDesktopElements(IEnumerable`1 desktopElements)+MoveNext() in D:\a\_work\1\s\src\Automation\TargetElementLocator.cs:line 31
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source)
   at Axe.Windows.Automation.SnapshotCommand.Execute(Config config, IScanTools scanTools) in D:\a\_work\1\s\src\Automation\SnapshotCommand.cs:line 43
   at Axe.Windows.Automation.Scanner.<>c__DisplayClass7_0.<ExecuteScan>b__0() in D:\a\_work\1\s\src\Automation\Scanner.cs:line 78
   at Axe.Windows.Automation.ExecutionWrapper.ExecuteCommand[T](Func`1 command) in D:\a\_work\1\s\src\Automation\ExecutionWrapper.cs:line 23 ---> System.ArgumentException: The argument must be a non-empty IEnumerable (Parameter 'desktopElements')
```

The issue originates from the fact that calling `UIAutomation.FindAll` in `A11yAutomation.cs`, line 69, returns no elements. This is a regression introduced by #676 since behavior was changed to filter elements using `TreeScope.TreeScope_Children` instead of the previous scope `TreeScope.TreeScope_Descendants`. This regression only seems to occur with UWP apps, WPF, Winform and WinUI 3 Desktop apps do not seem to exhibit this behavior and can be scanned just fine.

**To Reproduce**

1. Download WinUI 2 Gallery from Store
2. Run the following test:
```
var processes = Process.GetProcessesByName("AppUIBasics");
Assert.IsTrue(processes.Length > 0);

var config = Config.Builder.ForProcessId(processes[0].Id).WithMultipleScanRootsEnabled().Build();
var scanner = ScannerFactory.CreateScanner(config);
var testResult = scanner.ScanAll();
```

## Expected behavior
The scan should succeed normally.

## Actual behavior

Scanning UWP apps is not possible anymore.

## Desktop (please complete the following information):

 - OS: Windows 11 21H2 Build 22000.795
 - Accessibility Insights for Windows Version: N/A
 - Target Application: WinUI 2 Gallery, Files
 - Target Application Version: N/A

## Additional context
I would be more than happy to help fix this, however it seems that this is an OS issue since the UIAutomation layer does not give us the correct elements. A possible solution would be to have a fallback to return the first descendant in single scan mode. In multi scans, the list of descendants cannot be used since it lists all elements, not just the windows.

**Comment by RobGallo** [comment]
@chingucoding 
I think the most reliable way to solve this problem is probably not to call FindAll, but to use an IUIAutomationTreeWalker instead. How do you feel about giving that a try? If you want, I may be able to take a look in the coming days.

**Comment by chingucoding** [comment]
Sure, I can give that a try. Can you point me to an example of that or the general idea of how this could work @DaveTryon ?

**Comment by DaveTryon** [comment]
Hi, @chingucoding!

I opened WinUI 2 Gallery in AIWin, and there are 2 processes here. The topmost process (the child of the desktop) is named `ApplicationFrameHost.exe`. This process contains the system menu and the minimize/maximize/close buttons.

The second process, `AppUIBasics.exe` has a window that is a **child** of window owned by the `ApplicationFrameHost.exe` process. This the process that we're looking for. As pointed out earlier, we broke this in #676, where we limited the search to direct children of the desktop, rather than allowing the search to go arbitrarily deep.

The original code worked because it did a deep dive, effectively ignoring this note from the [remarks section of the API documentation](https://docs.microsoft.com/en-us/windows/win32/api/uiautomationclient/nf-uiautomationclient-iuiautomationelement-findfirst#remarks):

> When searching for top-level windows on the desktop, be sure to specify [TreeScope_Children](https://docs.microsoft.com/en-us/windows/desktop/api/uiautomationclient/ne-uiautomationclient-treescope) in the scope parameter, not [TreeScope_Descendants](https://docs.microsoft.com/en-us/windows/desktop/api/uiautomationclient/ne-uiautomationclient-treescope). A search through the entire subtree of the desktop could iterate through thousands of items and lead to a stack overflow.

So, to locate the process in question, we need to be able to do the following:
1. Go at at least 2 levels of depth
2. Look for elements that are not the first enumerated element
3. Still be reasonably performant

I wasn't able to locate sample code that does this, so I coded something up. I've created a [sample PR](https://github.com/DaveTryon/axe-windows/pull/45) of what this _might_ look like. It works locally for both UWP and non-UWP apps. I'm pretty sure that it's leaking memory from not releasing the items in `nonMatchingElements`. Hopefully that should be fairly easy to sort out, though.
